proxy/apifilterproxy/filters/auth/rancher_token_validate_filter.go

```diff
@@ -0,0 +1,289 @@
+package auth
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/filters"
+	"github.com/rancher/websocket-proxy/proxy/apifilterproxy/model"
+)
+
+const (
+	name = "authTokenValidator"
+)
+
+//AuthorizeData is for the JSON output
+type AuthorizeData struct {
+	Message string `json:"message,omitempty"`
+}
+
+//MessageData is for the JSON output
+type MessageData struct {
+	Data []interface{} `json:"data,omitempty"`
+}
+
+func init() {
```

Comment:
It isn't any more pluggable by using an init() function. You still have to load it from another package. Each handler package can still have NewHandler() function that does the same loading from environment variables. The change will just be to call NewHandler explicitly instead of doing `import _ github.com/blah/blah`